### DISCUSSION TITLE
c++ std::polar equivalent

### DISCUSSION THREAD

**Discussion by jecampagne** [jax-ml/jax#13630] [Q&A]
Hello,
What would be the best equivalent to the C++ std::polar function based in jax/lax/numpy function (ie seems that polar is not (yet) implemented)?
Thanks

**Comment by jakevdp** [comment]
Hi - I don't think JAX (or numpy) has any built-in function that's equivalent to `std::polar`, but it seems like it wouldn't be that hard to define from scratch, for example:
```python
def polar(rho, theta):
  return rho * (jnp.cos(theta) + 1j * jnp.sin(theta))
```

**Reply by jecampagne** [reply]
Yes that's what I though and I was wandering if there are clever implementation (note that std::polar states that it is 4.5x faster than the std:exp usage)...
Thanks

**Reply by hawkinsp** [reply]
I'm pretty sure Jake's suggestion is exactly how the C++ standard library implements it, e.g., here are two implementations:

* libstdc++: https://github.com/gcc-mirror/gcc/blob/95dc11475dac06b5eecd904079de8aa94827a36a/libstdc%2B%2B-v3/include/std/complex#L961
* libc++: https://github.com/llvm-mirror/libcxx/blob/78d6a7767ed57b50122a161b91f59f19c9bd0d19/include/complex#L994
